src/tools/debugger/mhpdbgc.c

```diff
@@ -1511,54 +1588,59 @@ void mhp_cmd(const char * cmd)
 
 static void mhp_print_ldt(int argc, char * argv[])
 {
+  static char lastldt[32];
   static char buffer[0x10000];
-  unsigned int *lp, *lp_=(unsigned int *)dpmi_get_ldt_buffer();
-  unsigned int base_addr, limit;
-  int type, type2, i;
+  unsigned int *lp, *lp_;
+  unsigned int base_addr, limit, i, lines=0;
+  int type, type2;
   unsigned int seg;
-  int lines=0, cache_mismatch;
+  int cache_mismatch;
   enum{Abit=0x100};
 
-  if (get_ldt(buffer) < 0) {
-    mhp_printf("error getting ldt\n");
+  if (!check_for_stopped())
+    return;
+
+  if (argc > 2 && !getval_ui(argv[2], 0, &lines)) {
+    mhp_printf("invalid number of lines '%s'\n", argv[2]);
     return;
   }
 
   if (argc > 1) {
-     if (IN_DPMI && isalpha(argv[1][0])) {
-       regnum_t rnum;
+    regnum_t rnum;
+
+    if (decode_symreg(argv[1], &rnum, NULL)) {
+      if (!IN_DPMI) {
+        mhp_printf("register values only valid in DPMI\n");
```

Comment:
This is quite bad, because unless you
are "stopped", IN_DPMI will change many
times a second. This just is not going to work.
I think we need to remove the reg option.

Reply:
Several lines above we return if not 'stopped'. does that make it alright, or shall I just rip it out?

Reply:
Ah, so now you do entire printing under stop...
Well I don't think this is a good idea because
for example you can dump mem while running -
its all fine unless you start fiddling with regs,
in which case you need this `IN_DPMI` thingy.
So I really think this makes the whole thing
inconsistent and should be removed. We then
don't need "stopped", which is a bit of a relief.